HIRS_Utils/src/main/java/hirs/data/persist/tpm/PcrInfoShort.java

```diff
@@ -235,6 +255,32 @@ public final int hashCode() {
         return computedDigest;
     }
 
+    /**
+     * Calculates the digest of the PCR values the same way a TPM computes the digest contained in
+     * the quote. Useful for TPM appraisal and for ensuring the digest of the collected PCR values
+     * match the digest in the quote.
+     *
+     * @param messageDigest message digest algorithm to use
+     * @return byte array containing the digest
+     */
+    public final byte[] getCalculatedDigestTpmV2p0(final MessageDigest messageDigest) {
```

Comment:
This could be a `private` method as we could force the user to go through the more general `getCalculatedDigest` method. I could also see leaving it `public` to let a user choose the algorithm for TPM 2.0. That being said, you could just overload the `getCalculatedDigest` and write another version of that that accepts an argument for MessageDigest which it can then pass through to this as a `private` method.

Reply:
fixed